### PR TITLE
Legend tuple handler improve

### PR DESCRIPTION
**Pull request by tacaswell**
This needs a bit more attention, but useful PR (I came up with a much-less clean way of doing [the same thing](https://stackoverflow.com/questions/18007022/two-unique-marker-symbols-for-one-legend/18007565#18007565))

  - needs better doc
  - needs maybe better kwarg name

replaces and closes #2034

### CODE REVIEW

**Comment by tacaswell** [comment]
@WeatherGod Updated

so

File: examples/pylab_examples/legend_demo5.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ind = [1, 2, 3]
+pos1 = [1, 3, 2]
+neg1 = [2, 1, 4]
+width = [0.5, 0.5, 0.5]
+
+rpos1 = ax2.bar(ind, pos1, width=0.5, color='k', label='+1')
+rneg1 = ax2.bar(ind, neg1, width=0.5, color='w', hatch='///', label='-1')
+
+l = ax2.legend([(rpos1, rneg1)], ['Test'],
+               handler_map={(rpos1, rneg1): HandlerTuple(ndivide=0, pad=0.)})
```
Comment (WeatherGod) [inline examples/pylab_examples/legend_demo5.py]: dictionary comprehensions weren't added until py2.7

**Comment by WeatherGod** [comment]
flash from the past!

Need at least one unit test and a what's new entry. I also don't think the example is all that clear. It could benefit greatly from some comments and/or a top-level docstring.

File: examples/pylab_examples/legend_demo5.py (hunk 1 above)
Comment (OceanWolf) [inline examples/pylab_examples/legend_demo5.py]: Hmm, why didn't Travis pick that up?  I thought it went through all examples...

Comment (WeatherGod) [inline examples/pylab_examples/legend_demo5.py]: It goes through the examples as part of the doc build, which is probably
not py2.6.

**Comment by tacaswell** [comment]
This is not at the top of my to-do queue, if anyone wants to take over this PR, please do!

**Comment by three-comrades** [comment]
This PR can be closed (see #5617).

**Pull request closed without merging**